tests/exporters/test_cds.py

```diff
@@ -97,3 +118,35 @@ def test_user_defined_selection_requests(self, cdsapi_mock):
         for key, val in expected_selection_request.items():
             default_val = default_selection_request[key]
             assert default_val == val, f'For {key}, expected {val}, got {default_val}'
+
+    @patch('cdsapi.Client')
+    def test_break_up(self, cdsapi_mock, tmp_path):
+        cdsapi_mock.return_value = Mock()
+        exporter = ERA5Exporter(tmp_path)
+
+        user_defined_arguments = {
+            'year': [2019, 2018],
+            'day': [1, 2],
+            'month': [4, 5],
+            'time': [0]
+        }
+
+        output_paths = exporter.export('precipitation',
+                                       dataset='era5',
+                                       granularity='hourly',
+                                       selection_request=user_defined_arguments,
+                                       break_up=True)
+
+        raw_folder = tmp_path / 'raw'
+        expected_paths = [
```

Comment:
@gabrieltseng so how does this `@patch('cdsapi.Client')` `Mock` test thing actually create the .nc files? Surely these are only the result of actually calling the API?

Reply:
No `.nc` files are actually created. The test only checks that the correct filenames were created (broken up by month). If those filenames are made, the correct API requests should have been made (also broken up by months), since, the selection request is used to make the filename.

Reply:
No .nc files are actually created. The test only checks that the correct filenames were created (broken up by month). If those filenames are made, the correct API requests should have been made (also broken up by months), since, the selection request is used to make the filename.